ecs/task.go

```diff
@@ -210,15 +210,15 @@ func (ecs *ECS) DescribeTasks(taskIds []string) []Task {
 			DesiredStatus: aws.StringValue(t.DesiredStatus),
 			LastStatus:    aws.StringValue(t.LastStatus),
 			Memory:        aws.StringValue(t.Memory),
-			TaskId:        taskId,
+			TaskId:        taskID,
 			StartedBy:     aws.StringValue(t.StartedBy),
 		}
 
 		taskDefinition := ecs.DescribeTaskDefinition(aws.StringValue(t.TaskDefinitionArn))
-		task.Image = aws.StringValue(taskDefinition.ContainerDefinitions[0].Image)
-		task.TaskRole = aws.StringValue(taskDefinition.TaskRoleArn)
+		task.Image = aws.StringValue(taskDefinition.TaskDefinition.ContainerDefinitions[0].Image)
+		task.TaskRole = aws.StringValue(taskDefinition.TaskDefinition.TaskRoleArn)
```

Comment:
"taskDefinition.TaskDefinition.foo" looks ugly.  :/

I see why you you're doing it; the api call DescribeTaskDefinition returns more than the task definition. https://docs.aws.amazon.com/AmazonECS/latest/APIReference/API_DescribeTaskDefinition.html

Maybe rename taskDefinition because it _is_ more than the task definition?  Unsure...

dtd := ecs.DescribeTaskDefinition(aws.StringValue(t.TaskDefinitionArn)) //returns more than task definition

Reply:
Yes, a real pain.  Thanks AWS! :)  Agree it looks bad...I'll rename it.